### PR TITLE
Fix Issue 17586 - Overriding a deprecated base class function gives no warning

### PR DESCRIPTION
**Pull request by RazvanN7**

### CODE REVIEW

**Review by PetarKirov — APPROVED** [review]

File: src/ddmd/dsymbolsem.d [hunk 1]
```diff
@@ -4567,6 +4567,10 @@ private extern(C++) final class DsymbolSemanticVisitor : Visitor
                             goto Lintro;
                     }
 
+                    if (fdv.isDeprecated)
+                        deprecation(funcdecl.loc, "%s is overriding the deprecated method %s",
```
Comment (JinShil) [inline src/ddmd/dsymbolsem.d]: Please add tick marks (`) around the %s

**Review by JinShil — CHANGES_REQUESTED** [review]

File: src/ddmd/dsymbolsem.d (hunk 1 above)
Comment (RazvanN7) [reply src/ddmd/dsymbolsem.d]: done, thanks!

**Review by wilzbach — APPROVED** [review]
Thanks!

**Comment by wilzbach** [comment]
The linked issue will automatically appear in the changelog (with title). Do you think this should be explained in a more detailed message?

**Comment by mathias-lang-sociomantic** [comment]
I think it wouldn't hurt to add a small explanation on every language change. I thought it was more common, but apparently we're not doing it, but we have a large audience, and communicating those changes properly (and their motivation) should be done upfront IMO. But we can just go with the status quo for now :)

**Pull request merged**